packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -86,6 +88,20 @@ export const usePostModalNavigation = (
     scrollPositionOnFeed.current = 0;
   };
 
+  useEffect(() => {
+    const routeHandler = (newRoute: string) => {
+      if (newRoute.includes('/posts/')) {
+        return;
+      }
+      onCloseModal();
+    };
+    router.events.on('routeChangeStart', routeHandler);
+
+    return () => {
+      router.events.off('routeChangeStart', routeHandler);
+    };
+  }, [router.events]);
```

Comment:
Lint might need to have the `onCloseModal` as a dependency. In that case, we might have to convert it to use `useCallback`.

Reply:
Yep, done now :)